File: src/main/java/com/snackgame/server/rank/applegame/domain/Season.java
package com.snackgame.server.rank.applegame.domain;

import java.time.LocalDateTime;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Entity
public class Season {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private LocalDateTime startedAt;
    private LocalDateTime endedAt;

    public Season(String name, LocalDateTime startedAt) {
        this.name = name;
        this.startedAt = startedAt;
    }

    public Season(String name, LocalDateTime startedAt, LocalDateTime endedAt) {
        this.name = name;
        this.startedAt = startedAt;
        this.endedAt = endedAt;
    }

    @Override
    public String toString() {
        return "Season{" +
               "name='" + name + '\'' +
               '}';
    }
}

File: src/main/java/com/snackgame/server/rank/domain/Season.kt
package com.snackgame.server.rank.domain

import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class Season(
    val name: String,
    val startedAt: LocalDateTime,
    val endedAt: LocalDateTime? = null,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
) {

    override fun toString(): String {
        return "Season(name='$name', startedAt=$startedAt, endedAt=$endedAt, id=$id)"
    }
}
